File: aat-lib/src/main/java/ch/bailu/aat_lib/map/layer/gpx/legend/LegendWalker.java
package ch.bailu.aat_lib.map.layer.gpx.legend;

import org.mapsforge.core.graphics.Paint;

import ch.bailu.aat_lib.gpx.GpxListWalker;
import ch.bailu.aat_lib.map.MapContext;
import ch.bailu.aat_lib.gpx.GpxList;
import ch.bailu.aat_lib.gpx.GpxSegmentNode;

public abstract class LegendWalker extends GpxListWalker {

    public LegendContext c;

    public void init(MapContext mc, Paint backgroundPaint, Paint framePaint) {
        c=new LegendContext(mc, backgroundPaint, framePaint);
    }


    @Override
    public boolean doList(GpxList track) {
        return
                track.getPointList().size() > 0 &&
                        c.isVisible(track.getDelta().getBoundingBox());
    }


    @Override
    public boolean doSegment(GpxSegmentNode segment) {
        return true;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/layer/gpx/legend/LegendWalker.kt
package ch.bailu.aat_lib.map.layer.gpx.legend

import ch.bailu.aat_lib.gpx.GpxList
import ch.bailu.aat_lib.gpx.GpxListWalker
import ch.bailu.aat_lib.gpx.GpxSegmentNode
import ch.bailu.aat_lib.map.MapContext
import org.mapsforge.core.graphics.Paint

abstract class LegendWalker : GpxListWalker() {
    @JvmField
    var legendContext: LegendContext? = null

    fun init(mcontext: MapContext, backgroundPaint: Paint, framePaint: Paint) {
        legendContext = LegendContext(mcontext, backgroundPaint, framePaint)
    }

    override fun doList(track: GpxList): Boolean {
        return track.pointList.size() > 0 && legendContext?.isVisible(track.getDelta().getBoundingBox()) == true
    }

    override fun doSegment(segment: GpxSegmentNode): Boolean {
        return true
    }
}
